File: app/local_plugins/llm_volcengine/android/src/main/kotlin/com/aiagent/llm_volcengine/LlmVolcenginePlugin.kt
package com.aiagent.llm_volcengine

import android.util.Log
import com.aiagent.plugin_interface.NativeServiceRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * LlmVolcenginePlugin — 火山引擎 Ark LLM Flutter 插件
 *
 * 注册 vendor="volcengine"（兼容旧数据 "doubao"），走 ark OpenAI 兼容接口
 * (POST /api/v3/chat/completions + Bearer ApiKey)。
 */
class LlmVolcenginePlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        listOf("volcengine", "doubao").forEach { vendor ->
            NativeServiceRegistry.registerLlm(vendor) { LlmVolcengineService() }
        }
        Log.d("LlmVolcenginePlugin", "Registered NativeLlmService for vendors: volcengine, doubao")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // no-op
    }
}

File: app/local_plugins/vendors/llm_volcengine/ios/Classes/LlmVolcenginePlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// Volcengine Ark LLM Flutter plugin entry.
///
/// Mirrors the Android `LlmVolcenginePlugin`: registers vendors
/// "volcengine" and "doubao" against `LlmVolcengineService`. The service
/// talks to Ark's OpenAI-compatible `chat/completions` endpoint
/// (https://ark.cn-beijing.volces.com/api/v3 by default).
public class LlmVolcenginePlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.llm_volcengine", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        let vendors = ["volcengine", "doubao"]
        for vendor in vendors {
            NativeServiceRegistry.shared.registerLlm(vendor) { LlmVolcengineService() }
        }
        os_log("Registered NativeLlmService for vendors: %{public}@",
               log: log, type: .debug, vendors.description)
    }
}
